### ISSUE TITLE
zlib must be installed

### ISSUE THREAD

**Issue by JeWaVe** [mikael-s-persson/templight#89]
Building with VS2022 I get errors relative to LLVMSUpport 
templight needs llvm::zlib enabled 
I managed to fix the error by install zlib, disabling UNISTD in zlib.conf 

This should probably be added to documentation

**Comment by flsmith** [comment]
@JeWaVe could you possibly provide some more details? I'm having similar issues, though I'm not building in VS2022. Is the zlib.conf something specific to that? I can't find such a file.

For reference, I'm trying to build using `cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS=clang -DLLVM_ENABLE_ZLIB=ON`, and the system has `zlib` already installed.

**Comment by Witcher01** [comment]
I've also been unable to build this due to the same error:

```
/home/thomas/sources/llvm-project/clang/tools/templight/lib/TemplightProtobufWriter.cpp: In member function ‘std::string clang::TemplightProtobufWriter::printTemplateName(const std::string&)’:
/home/thomas/sources/llvm-project/clang/tools/templight/lib/TemplightProtobufWriter.cpp:243:15: error: ‘llvm::zlib’ has not been declared
  243 |     if (llvm::zlib::compress(llvm::StringRef(Name), CompressedBuffer)) {
      |               ^~~~
```

Building on Arch Linux (Linux 5.19.10-arch1-1) with cmake 3.24.2 and zlib 1.2.12-2.
Neither
```
cmake -DLLVM_ENABLE_PROJECTS=clang -DCMAKE_BUILD_TYPE=Debug ../llvm/
```
nor
```
cmake -DLLVM_ENABLE_PROJECTS=clang -DLLVM_ENABLE_ZLIB=ON -DCMAKE_BUILD_TYPE=Debug ../llvm/
```
worked for building.

@JeWaVe I'm not sure what `zlib.conf` file you mean. Can you elaborate?

@mikael-s-persson do you happen to know the cause of this issue?

**Comment by meow464** [comment]
Use LLVM 14.
LLVM 15 won't work because`llvm::zlib` [was moved](https://github.com/llvm/llvm-project/commit/ea61750c35a11140e9245bd9cbeb383c37f6e031) to `llvm::compression::zlib`.
LLVM 13 won't work for a [different reason](https://github.com/llvm/llvm-project/commit/89b57061f7b769e9ea9bf6ed686e284f3e55affe).

Clang gave a much more useful error than gcc and used a bit less ram too.
I might make a pull request once I fix this thing I'm working on.

**Comment by mikael-s-persson** [comment]
I think this should be fixed in #91
